libs/tendermint/mempool/clist_mempool.go

```diff
@@ -305,6 +309,22 @@ func (mem *CListMempool) TxsWaitChan() <-chan struct{} {
 //
 // Safe for concurrent use by multiple goroutines.
 func (mem *CListMempool) CheckTx(tx types.Tx, cb func(*abci.Response), txInfo TxInfo) error {
+	mem.peersTxCountMtx.Lock()
+	if len(txInfo.SenderP2PID) != 0 {
+		peerTxCount, ok := mem.peersTxCount[string(txInfo.SenderP2PID)]
+		if !ok {
+			peerTxCount = 0
+		}
+		if cfg.DynamicConfig.GetMaxTxLimitPerPeer() != 0 && peerTxCount >= cfg.DynamicConfig.GetMaxTxLimitPerPeer() {
```

Comment:
Is it better that if the GetMaxTxLimitPerPeer() == 0, then we do not need to lock any way?